File: ReactApp/src/Services/WorkService.js
import axios from 'axios'
const WORK_BASE_REST_API_URL = 'http://localhost:9000/api/work';

class WorkService{

    getAllWork(){
        return axios.get(WORK_BASE_REST_API_URL)
    }

    createWork(work){
        return axios.post(WORK_BASE_REST_API_URL, work)
    }

    getWorkById(workId){
        return axios.get(WORK_BASE_REST_API_URL + "/" + workId)
    }

    getWorkByUserId(userId){
        return axios.get(WORK_BASE_REST_API_URL + "/search/" + userId)
    }

    getWorkByIdName(idName){
        return axios.get(WORK_BASE_REST_API_URL+ "/search/idName/" + idName)
    }

    updateWork(workId, work){
        return axios.put(WORK_BASE_REST_API_URL + "/" + workId, work)
    }

    deleteWork(workId){
        return axios.delete(WORK_BASE_REST_API_URL + "/" + workId)
    }

    deleteWorkByUserId(userId){
        return axios.delete(WORK_BASE_REST_API_URL + "/delete/" + userId)
    }

}

export default new WorkService()

File: src/main/java/com/utsav/cms/service/WorkService.java
package com.utsav.cms.service;

import java.util.List;

import com.utsav.cms.model_bean.User;
import com.utsav.cms.model_bean.Work;

public interface WorkService {
	Work saveWork(Work work);
	List<Work> getAllWork();
	Work getWorkById(long id);
	List<Work> getWorkByUserId(User userId);
	List<Work> getWorkByIdName(String idName);
	Work updateWork(Work work, long id);
	String deleteWork(long id);
	String deleteWorkByUserId(User userId);

}
